### PR TITLE
gapic: wrap errors from REST better

### PR DESCRIPTION
**Pull request by software-dov**
If the http client encounters an error when making a requst, it uses
the response to build a googleapi.Error instead in order to better
preserve information and aid readability.

### CODE REVIEW

File: showcase/echo_test.go [hunk 1]
```diff
@@ -68,19 +69,25 @@ func TestEcho_error(t *testing.T) {
...
-		status, _ := status.FromError(err)
-		if status.Code() != val {
-			t.Errorf("%s Echo() errors with %d, want %d", typ, status.Code(), val)
+		if typ == "grpc" {
+			status, _ := status.FromError(err)
+			if status.Code() != val {
+				t.Errorf("%s Echo() errors with %d, want %d", typ, status.Code(), val)
+			}
+		} else {
+			const ERR_CODE int = 500 // Unavailable
```
Comment (noahdietz) [inline showcase/echo_test.go:82]: Use `StatusInternalServerError` from https://pkg.go.dev/net/http instead please. also 503 is unavailable, 500 is internal server error.

File: internal/gengapic/genrest.go [hunk 2]
```diff
@@ -845,8 +846,9 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	p("}")
 	p("defer httpRsp.Body.Close()")
 	p("")
-	p("if httpRsp.StatusCode != http.StatusOK {")
-	p("  return nil, fmt.Errorf(httpRsp.Status)")
+	p("err = googleapi.CheckResponse(httpRsp)")
+	p("if err != nil {")
```
Comment (noahdietz) [inline internal/gengapic/genrest.go:850]: Nit: Can you inline this please?

File: internal/gengapic/genrest.go [hunk 3]
```diff
@@ -608,8 +608,9 @@ func (g *generator) pagingRESTCall(servName string, m *descriptor.MethodDescript
 	p("  }")
 	p("  defer httpRsp.Body.Close()")
 	p("")
-	p("  if httpRsp.StatusCode != http.StatusOK {")
-	p(`    return nil, "", fmt.Errorf(httpRsp.Status)`)
+	p("  err = googleapi.CheckResponse(httpRsp)")
+	p("  if err != nil {")
```
Comment (noahdietz) [inline internal/gengapic/genrest.go:612]: Nit: Can you inline this please?

File: showcase/echo_test.go (hunk 1 above)
Comment (software-dov) [reply showcase/echo_test.go:82]: That is exactly what I was looking for. Thanks!

File: internal/gengapic/genrest.go (hunk 2 above)
Comment (software-dov) [reply internal/gengapic/genrest.go:850]: Done

File: internal/gengapic/genrest.go (hunk 3 above)
Comment (software-dov) [reply internal/gengapic/genrest.go:612]: Done.

**Review by noahdietz — APPROVED** [review]

**Pull request merged**